harness/tests/determined/common/experimental/test_experiment.py

```diff
@@ -101,7 +100,7 @@ def test_wait_raises_exception_when_experiment_is_paused(
 
 
 @responses.activate
-def test_list_trials_iterates_through_all_trials(
```

Comment:
I like the new test. Why did you delete the old ones? They still look relevant to me.

Reply:
small nit: in case you prefer the language, `test_list_trials_respects_max_results` seems a little more accurate to me. But I could also understand if you think it obscures some of the test's meaning, too.

Reply:
> Why did you delete the old ones?

goofed. added it back.

Reply:
`test_list_trials_respects_max_results` is a good addition, but I think I expect `test_list_trials_iterates_through_all_trials` to still be a valid test for when `max_results` isn't passed. Agree/disagree?

Reply:
makes sense. i should have treated `max_results` as a new feature, these tests are still valid.